File: app/src/main/java/com/platdmit/simplecloudmanager/domain/helpers/ActualApiKeyService.java
package com.platdmit.simplecloudmanager.domain.helpers;

import com.platdmit.simplecloudmanager.domain.models.UserAccount;

import io.reactivex.rxjava3.subjects.BehaviorSubject;

public interface ActualApiKeyService {
    void setActiveAccount(UserAccount activeAccount);
    void startAutoUpdate(UserAccount activeAccount);
    void startDemoMode();
    void stopAutoUpdate();
    BehaviorSubject<Boolean> getAccountStatus();
    String getApiKey();
}

File: app/src/main/java/com/platdmit/simplecloudmanager/domain/helpers/ActualApiKeyService.kt
package com.platdmit.simplecloudmanager.domain.helpers

import com.platdmit.simplecloudmanager.domain.models.UserAccount
import io.reactivex.rxjava3.subjects.BehaviorSubject

interface ActualApiKeyService {
    fun setActiveAccount(activeAccount: UserAccount)
    fun startAutoUpdate(activeAccount: UserAccount)
    fun startDemoMode()
    fun stopAutoUpdate()
    fun getAccountStatus(): BehaviorSubject<Boolean>
    var apiKey: String
}